### PR TITLE
Avoid multiple reload operations when configure multiple interfaces

### PR DESCRIPTION
**Pull request by nmeyerhans**
In the event that we're configuring several interfaces (e.g. an instance with several attached ENIs (re)boots), we would previously issue multiple approximately simultaneous `networkctl reload` commands. With this PR, we ensure that when we have multiple concurrent setup-policy-routes processes running, only one of them ends up issuing the reload command.

At a high level, we accomplish this by creating a directory in a well-known location, and creating a file in it. Then, before exiting, we remove the file we created and attempt to remove the directory. If the `rmdir` call succeeds, then we've determined that there were no other running setup-policy-routes processes and reload networkd if necessary. If `rmdir` fails, then we've determined that another setup-policy-routes process is running and exit, letting the other process handle the reload.

Naturally, there's subtlety involved in any sort of interprocess coordination like this. A SIGKILL to a setup-policy-routes process would prevent state cleanup, essentially leading to a deadlock in which nobody can reload networkd without manual cleanup, but otherwise I believe I've covered the failure cases. Careful review and plenty of questions would be appreciated, though.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: bin/setup-policy-routes.sh [hunk 1]
```diff
@@ -310,6 +310,34 @@ setup_interface() {
     echo $changes
 }
 
+# All instances of this process that may reconfigure networkd register
+# themselves as such. When exiting, they'll reload networkd only if
+# they're the registered process running.
+maybe_reload_networkd() {
+    rm -f /run/setup-policy-routes/$$
```
Comment (halfdime-code) [inline bin/setup-policy-routes.sh:317]: Would using /dev/shm as the root for all these files better serve us by eliminating the entire class of bugs related to deadlocking across a reboot?

Comment (halfdime-code) [inline bin/setup-policy-routes.sh:317]: Would using /dev/shm as the root for all these files better serve us by eliminating the entire class of bugs related to deadlocking across a reboot?

Comment (nmeyerhans) [reply bin/setup-policy-routes.sh:317]: `/run` is a ramdisk.

**Comment by nmeyerhans** [comment]
Rebased onto the latest main branch after merging the other open PRs. @halfdime-code are you good with this?

File: bin/setup-policy-routes.sh (hunk 1 above)
Comment (halfdime-code) [reply bin/setup-policy-routes.sh:352]: Well then. Never mind. ;-)

**Review by halfdime-code — APPROVED** [review]

**Pull request merged**